include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,324 @@
+	SOF_IPC4_GLB_NOTIFICATION = 27,
+	SOF_IPC4_GLB_MAX_IXC_MESSAGE_TYPE = 31
+};
+
+/* Message direction */
+enum sof_ipc4_msg_dir {
+	SOF_IPC4_MSG_REQUEST = 0,
+	SOF_IPC4_MSG_REPLY = 1,
+};
+
+enum sof_ipc4_pipe_line_state {
+	SOF_IPC4_PIPE_RESET = 2,
+	SOF_IPC4_PIPE_PAUSED = 3,
+	SOF_IPC4_PIPE_RUNNING = 4,
+	SOF_IPC4_PIPE_EOS = 5
+};
+
+/* global common ipc msg */
+#define SOF_IPC4_GLB_MSG_TARGET_SHIFT		30
+#define SOF_IPC4_GLB_MSG_TARGET_MASK		BIT(30)
+#define SOF_IPC4_GLB_MSG_TARGET(x)		((x) << SOF_IPC4_GLB_MSG_TARGET_SHIFT)
+
+#define SOF_IPC4_GLB_MSG_DIR_SHIFT		29
+#define SOF_IPC4_GLB_MSG_DIR_MASK		BIT(29)
+#define SOF_IPC4_GLB_MSG_DIR(x)			((x) << SOF_IPC4_GLB_MSG_DIR_SHIFT)
+
+#define SOF_IPC4_GLB_MSG_TYPE_SHIFT		24
+#define SOF_IPC4_GLB_MSG_TYPE_MASK		GENMASK(28, 24)
+#define SOF_IPC4_GLB_MSG_TYPE(x)		((x) << SOF_IPC4_GLB_MSG_TYPE_SHIFT)
+
+/* pipeline creation ipc msg */
+#define SOF_IPC4_GLB_PIPE_INSTANCE_SHIFT	16
+#define SOF_IPC4_GLB_PIPE_INSTANCE_MASK		GENMASK(23, 16)
+#define SOF_IPC4_GLB_PIPE_INSTANCE_ID(x)	((x) << SOF_IPC4_GLB_PIPE_INSTANCE_SHIFT)
+
+#define SOF_IPC4_GLB_PIPE_PRIORITY_SHIFT	11
+#define SOF_IPC4_GLB_PIPE_PRIORITY_MASK		GENMASK(15, 11)
+#define SOF_IPC4_GLB_PIPE_PRIORITY(x)		((x) << SOF_IPC4_GLB_PIPE_PRIORITY_SHIFT)
+
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE_SHIFT	0
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE_MASK		GENMASK(10, 0)
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE(x)		((x) << SOF_IPC4_GLB_PIPE_MEM_SIZE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_EXT_LP_SHIFT		0
+#define SOF_IPC4_GL_PIPE_EXT_LP_MASK		BIT(0)
+#define SOF_IPC4_GL_PIPE_EXT_LP(x)		((x) << SOF_IPC4_GL_PIPE_EXT_LP_SHIFT)
+
+/* pipeline set state ipc msg */
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT	24
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_MASK	GENMASK(28, 24)
+#define SOF_IPC4_GL_PIPE_STATE_TYPE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_ID_SHIFT		16
+#define SOF_IPC4_GL_PIPE_STATE_ID_MASK		GENMASK(23, 16)
+#define SOF_IPC4_GL_PIPE_STATE_ID(x)		((x) << SOF_IPC4_GL_PIPE_STATE_ID_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_SHIFT		0
+#define SOF_IPC4_GL_PIPE_STATE_MASK		GENMASK(15, 0)
+#define SOF_IPC4_GL_PIPE_STATE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_SHIFT)
+
+enum sof_ipc4_sampling_frequency {
+	SOF_IPC4_FS_8000HZ	= 8000,
+	SOF_IPC4_FS_11025HZ	= 11025,
+	SOF_IPC4_FS_12000HZ	= 12000, /* Mp3, AAC, SRC only. */
+	SOF_IPC4_FS_16000HZ	= 16000,
+	SOF_IPC4_FS_18900HZ	= 18900, /* SRC only for 44100 */
+	SOF_IPC4_FS_22050HZ	= 22050,
+	SOF_IPC4_FS_24000HZ	= 24000, /* Mp3, AAC, SRC only. */
+	SOF_IPC4_FS_32000HZ	= 32000,
+	SOF_IPC4_FS_37800HZ	= 37800, /* SRC only for 44100 */
+	SOF_IPC4_FS_44100HZ	= 44100,
+	SOF_IPC4_FS_48000HZ	= 48000, /* Default. */
+	SOF_IPC4_FS_64000HZ	= 64000, /* AAC, SRC only. */
+	SOF_IPC4_FS_88200HZ	= 88200, /* AAC, SRC only. */
+	SOF_IPC4_FS_96000HZ	= 96000, /* AAC, SRC only. */
+	SOF_IPC4_FS_176400HZ	= 176400, /* SRC only. */
+	SOF_IPC4_FS_192000HZ	= 192000, /* SRC only. */
+	SOF_IPC4_FS_INVALID
```

Comment:
so the value of 192001 is invalid? This should be an explicit value really, it makes no sense to mix explicit values and implicit increments.